assignment/tool/src/webapp/js/videoRecorder.js

```diff
@@ -0,0 +1,74 @@
+function getBase64(file) {
+  return new Promise((resolve, reject) => {
+    const reader = new FileReader();
+    reader.readAsDataURL(file);
+    reader.onload = () => resolve(reader.result);
+    reader.onerror = (error) => reject(error);
+  });
+}
+
+let recorder;
+let player = document.getElementById('submission-recorder');
+
+function captureCamera(callback) {
+    navigator.mediaDevices.getUserMedia({ audio: true, video: true }).then(function(camera) {
+      callback(camera);
+    }).catch(function(error) {
+      alert('Unable to capture your camera. Please check console logs.');
+      console.error(error);
+    });
+}
+function stopRecordingCallback() {
+    player.src = player.srcObject = null;
+    player.muted = false;
+    player.volume = 1;
+    player.src = URL.createObjectURL(recorder.getBlob());
+
+    // TODO: Ideally we want a different mechanism to send the file to the backend as it's sent by a classic form post.
+    let fileObject = new File([recorder.getBlob()], "video.webm", {type: 'video/webm'});
+    getBase64(fileObject).then(
+      (data) => document.getElementById('videoResponse').value = data
+    );
+
+    recorder.camera.stop();
+    recorder.destroy();
+    recorder = null;
+}
+
+document.getElementById('btn-start-recording').onclick = function() {
+    player.style.display = 'block';
+    let submissionPlayers = document.querySelectorAll('#submission-player');
+    if (submissionPlayers) {
+      submissionPlayers.forEach( (submissionPlayer) => submissionPlayer.style.display = 'none');
+    }
+    this.disabled = true;
+    captureCamera(async function(camera) {
```

Comment:
I don't think you need async here. You're already running this in a then from a promise. I may be wrong though :)